cluster-autoscaler/provisioningrequest/checkcapacity/condition_test.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package checkcapacity
+
+import (
+	"testing"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/apis/autoscaling.x-k8s.io/v1beta1"
+	"k8s.io/autoscaler/cluster-autoscaler/provisioningrequest/provreqwrapper"
+)
+
+func TestBookCapacity(t *testing.T) {
+	tests := []struct {
+		name         string
+		prConditions []v1.Condition
+		want         bool
+	}{
+		{
+			name: "Expired",
+			prConditions: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1beta1.Expired,
+					Status: v1.ConditionTrue,
+				},
+			},
+			want: false,
+		},
+		{
+			name: "Failed",
+			prConditions: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1beta1.Failed,
+					Status: v1.ConditionTrue,
+				},
+			},
+			want: false,
+		},
+		{
+			name: "empty conditions",
+			want: false,
+		},
+		{
+			name: "Capacity found and provisioned",
+			prConditions: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1beta1.Provisioned,
+					Status: v1.ConditionTrue,
+				},
+			},
+			want: true,
+		},
+		{
+			name: "Capacity is not found",
+			prConditions: []v1.Condition{
+				{
+					Type:   v1beta1.CapacityFound,
+					Status: v1.ConditionFalse,
+				},
+			},
+			want: false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			pr := provreqwrapper.NewV1Beta1ProvisioningRequest(
+				&v1beta1.ProvisioningRequest{
+					Spec: v1beta1.ProvisioningRequestSpec{
+						ProvisioningClassName: v1beta1.ProvisioningClassCheckCapacity,
+					},
+					Status: v1beta1.ProvisioningRequestStatus{
+						Conditions: test.prConditions,
+					},
+				}, nil)
+			got := bookCapacity(pr)
+			if got != test.want {
+				t.Errorf("Want: %v, got: %v", test.want, got)
+			}
+		})
+	}
+}
+
+func TestSetCondition(t *testing.T) {
```

Comment:
I would also add a test case where the is an existing unknown condition and we just ignore it.